src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromArray = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && typeof this[i] !== 'undefined') {
+        stringFromArray += `${this[i]}`;
+
+        return stringFromArray;
+      } else if (i === this.length - 1) {
+        return stringFromArray;
+      }
+
+      if (typeof this[i] !== 'undefined' && typeof this[i] !== 'object') {
+        stringFromArray += `${this[i]}${separator}`;
+        continue;
+      }
+      stringFromArray += `${separator}`;
+    }
```

Comment:
Avoid using `break`/`continue` unless not avoidable, it make code really hard to read. Again, we can handle separators once, no need to add it to every case

```suggestion
      if (typeof this[i] !== 'undefined' && typeof this[i] !== 'object') {
        stringFromArray += `${this[i]}`;
      }
      if (i !== this.length - 1) {
        stringFromArray += `${separator}`;
      }
    }
```